swap/htlc.go

```
@@ -133,6 +145,12 @@ func NewHtlc(version ScriptVersion, cltvExpiry int32,
 			cltvExpiry, senderKey, receiverKey, hash,
 		)
 
+	case HtlcV3:
```

For `HtlcV1` and `HtlcV2` we should return an error if the `sharedKey != nil`

---

An example of how to standardize: 

```
        // ErrNoSharedKey is returned when a script version does not support
	// use of a shared key.
	ErrNoSharedKey = errors.New("shared key not supported for script " +
		"version")

	// ErrSharedKeyRequired is returned when a script version requires a
	// shared key.
	ErrSharedKeyRequired = errors.New("shared key required")
```